[skip ci] Fix unit tests on macOS (And probably Unix)

Fesaa opened the pull request:
This PR fixes all but 7 tests on macOS. The remaining tests are in DirectoryServiceTest

<img width="1323" alt="image" src="https://github.com/user-attachments/assets/c34b4b7d-154d-45b4-9170-a0ef34ddbea7" />

Top one fails because `\` is valid in a filename. And the othesr fails because `C:/` is hardcoded, I could change them to construct it from Root +, but the test already has them mixed; so I think that would  change the test too much. 
Unsure what to do with them.


# Fixed
- Fixed several tests not running on macOS/Unix because of `C:/` as root
- Fixed a test involving time not passing on timezones > UTC because of how the setter works in the MockFileData

# Changed
- Unified several test classes to use the new AbstractFsTest to simplify setup
- Skip the IsDriveMounted tests on macOS/Unix as the root `/` is always mounted, and `C:/` will throw
- Change some tests to test against the CurrentLocale instead as the result is user facing


### Notes
I've had a friend run the tests on their Windows PC, and everything passes for them. However, they're on the same timezone as me. So would probably need to have someone tests whose on a timezone "earlier" than UTC.

Fesaa committed:
Fix CacheHelpertests on MacOS and TimeZones > UTC

MockFileData nomalized the LastWriteTime to UTC

Fesaa committed:
Fix bunch of issues with using C:/ as root

Fesaa committed:
Fix HumanReadableBytes tests on different Locale's

Fesaa committed:
Merge file system sets into AbstractFsTest

Only DirectoryService tests are failing now

Fesaa committed:
Skip DriveMounted tests on non Windows, change to Root where it makes sense

Fesaa committed:
Merge branch 'develop' into fix/unit-tests-macos

Fesaa committed:
Normalize root path

majora2007 reviewed (approved):
Looks good. I'll merge this after i do the hotfix.

majora2007 commented:
Actually builds on GA are failing now, we need to address that.

Fesaa commented:
~~This should fix it, but I'm unsure why~~

I'll be home again tomorrow, and will fix them then as I don't have a Windows PC here.

Fesaa committed:
Fix tests on Windows

Why does this fail when the root is normalized?

Fesaa committed:
Actually fix tests on Windows

It seems that the MockFileSystem defaults to C:/ as root when none is given. While GitHub actions (apparently) don't run on the C drive luckily neither are my GitHub repo's on my Windows PC.

This also updates the other DateTime tests to be inline with the UTC fix to ensure they're testing the right thing.

Please work now

majora2007 commented on API.Tests/Services/DirectoryServiceTests.cs:
```diff
@@ -373,9 +375,15 @@ public void CopyDirectoryToDirectory_ShouldCopyAllFileAndNestedDirectoriesOver()
     {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return;
```
Let's put a logger to indicate this test is not running.

majora2007 reviewed (approved):
I'm fine to merge then add the logging statement afterwards, or you can do it (on all applicable tests) and I will merge.

Fesaa committed:
Log when skipping these tests
